pkg/util/image.go

```diff
@@ -37,3 +38,29 @@ func ParseImageSpec(image string) (string, string, error) {
 	}
 	return server, repo, err
 }
+
```

Comment:
everything below in this PR (from `pkg/util/`) are just util functions that were living in `pod_controller.go`; I moved them to util package as well as tests to slim down `pod_controller.go`